### DISCUSSION TITLE
Installare HACS nell'immagine per VM [RISOLTO]

### DISCUSSION THREAD

**Discussion by NdK73** [indomus/forum#1803] [HUB: Home Assistant]
Ciao a tutti.

E' da ieri che cerco di installare HACS ma continua a non vederlo.

Il sistema base è una VM (sto facendo delle prove) installata seguendo le [istruzioni ufficiali](https://www.home-assistant.io/installation/generic-x86-64).

Mi sono collegato via ssh (abilitando e configurando l'add-on "SSH & Web Terminal") e lanciando il comando raccomandato per HACS (**PESSIMA** pratica wget + pipe, ma tanto era solo per un test!), poi ho riavviato HA e ancora non si vede nulla. :(

Ho provato anche con la webshell e mi dice che c'è già, lo sostituisce, riavvio e di nuovo nulla. Mi sento molto stupido. Sicuramente ho perso uno step importante, ma non riesco a capire quale...

Qualche idea?

Grazie

**Comment by snakuzzo** [comment]
1. Il titolo è sbagliato. Non è che installi HACS in un'immagine
2. la pessima pratica di cui parli, non è affatto pessima
3. Se non vai a vedere il log di Home Assistant stiamo parlando di aria fritta. Quando aggiungi un custom component (HACS è un custom component), quando avii homeassistant nel log ti appare un warning. Se non vedi quel warning, vuol dire che hai sbagliato qualcosa.

**Reply by NdK73** [reply]
1) beh, veniva un po' troppo lungo dire "nel disco creato convertendo in raw l'immagine QCOW2 fornita per l'installazione in VM" :)
2) dal punto della sicurezza operativa, si: abitua l'utente a scaricare da un sito un file "qualsiasi" e ad eseguirlo come root senza nessun controllo... sicuramente comodo, ma dal punto di vista della sicurezza non ci siamo proprio, anche considerando quanto sono diventati comuni i supply chain attack -- almeno scaricare il file, controllarlo e poi eseguirlo se è OK... non il massimo ma almeno si è più consapevoli di cosa si sta facendo
3) sapere cosa guardare e in quale log di HA sarebbe un ottimo punto di partenza... Comunque ho trovato solo questo riferimento ad hacs:
```
2022-08-04 13:08:04.853 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
```

Quindi pare trovarlo, ma poi non lo vedo ne sotto i componenti aggiuntivi (Settings -> Add-ons , neppure sotto "Add-on store"), ne nella barra laterale.
Le istruzioni ufficiali non aiutano: dopo aver lanciato l'installer dice di riavviare HA (ho anche fatto reboot della VM per essere sicuro, e cancellato la cache del browser) e poi andare sotto "Configuration -> Integrations" che non esiste più da parecchie versioni... Ovviamente non capisco dove andare per attivare HACS e configurare l'accesso a GitHub...

**Reply by NdK73** [reply]
Urgh! Trovato, finalmente.
Effettivamente c'era un misunderstanding: continuavo a cercare sotto gli add-on, ma era sotto "Devices & Services" (Dispositivi e Servizi) -> "Add integration". Uff... Magari sarà utile a qualcun altro.
Risolto.
Grazie.

**Reply by snakuzzo** [reply]
Tralasciando i punti 1 e 2 sui quali si potrebbe discutere, ma si andrebbe off topic, il custom component è caricato e basta smanettarci due secondi per capire dove andare. Il riavvio della VM a cosa dovrebbe servire ? :) Ti basta riavviare HA.

Innanzi tutto, hai letto la [nostra guida](https://indomus.it/guide/installare-hacs-home-assistant-community-store-sul-proprio-hub/) ? Te lo chiedo perché c'è scritto.

![aaa](https://user-images.githubusercontent.com/7205371/182871112-bfe8b2d8-bbe3-47c2-9e68-3f43f900b956.png)


Il sito ufficiale si basa sulla GUI di una vecchia versione di HA dove era nel menu "integrations". Da qualche versione è stato riorganizzato, ma non stiamo parlando di chissà cosa. Come dicevo prima, ci vogliono due secondi per trovare l'interfaccia giusta.

**Reply by snakuzzo** [reply]
> Urgh! Trovato, finalmente. Effettivamente c'era un misunderstanding: continuavo a cercare sotto gli add-on, ma era sotto "Devices & Services" (Dispositivi e Servizi) -> "Add integration". Uff... Magari sarà utile a qualcun altro. Risolto. Grazie.

credimi, lo sanno tutti.
E comunque HACS non è un add-on, è un custom component...c'è una bella differenza

**Reply by NdK73** [reply]
Yup. A posteriori è ovvio :) A volte capita di confondere due termini quando non si è troppo pratici e insistere selezionando l'opzione sbagliata senza notare quella giusta (e con l'avanzare dell'età succede sempre più spesso...).
Non avevo pensato alla vostra guida perché dovevo settarlo in inglese, e pareva fosse più semplice seguire quella in inglese per non dover "tradurre" al volo.

**Reply by snakuzzo** [reply]
Scusa, non capisco la cosa che hai detto sulla "lingua"...che c'entra ?
Sono quattro cavolate in croce, non penso ci sia bisogno di una traduzione.
Boh...a volte resto basito (F4 cit.). Non è per fare polemica eh, ma hai fatto tutto quel discorso (leggermente esagerato) su una wget e poi ti sei perso per così poco ? 
Hai scritto "si abitua l'utente a scaricare da un sito un file "qualsiasi" e ad eseguirlo come root senza nessun controllo" ... ma dove ?! :D
Non è che i comandi vengono eseguiti come `root` a pene di segugio...va fatto "quando serve".

Vabbè...comunque l'importante è che tu abbia risolto.
